app/bot.go

```diff
@@ -198,6 +198,108 @@ func (a *App) disableUserBots(userId string) *model.AppError {
 	return nil
 }
 
+func (a *App) notifySysadminsBotOwnerDeactivated(userId string) *model.AppError {
+	perPage := 1000
+	options := &model.BotGetOptions{
+		OwnerId:        userId,
+		IncludeDeleted: false,
+		OnlyOrphaned:   false,
+		Page:           0,
+		PerPage:        perPage,
+	}
+	userBots, err := a.GetBots(options)
+	if err != nil {
+		return err
+	}
+
+	// user does not own bots
+	if len(userBots) == 0 {
+		return nil
+	}
+
+	// user being disabled
+	user, err := a.GetUser(userId)
+	if err != nil {
+		return err
+	}
+
+	// get sysadmins
+	userOptions := &model.UserGetOptions{
+		Page:     0,
+		PerPage:  perPage,
+		Role:     model.SYSTEM_ADMIN_ROLE_ID,
+		Inactive: false,
+	}
+	sysAdmins, err := a.GetUsers(userOptions)
+	if err != nil {
+		return err
+	}
+
+	// for each sysadmin, notify user that owns bots was disabled
+	for _, sysAdmin := range sysAdmins {
+		channel, appErr := a.GetOrCreateDirectChannel(sysAdmin.Id, sysAdmin.Id)
+		if appErr != nil {
+			return appErr
+		}
+
+		post := &model.Post{
+			UserId:    sysAdmin.Id,
+			ChannelId: channel.Id,
+			Message:   a.getDisableBotSysadminMessage(user, userBots),
+			Type:      model.POST_SYSTEM_GENERIC,
+		}
+
+		_, appErr = a.CreatePost(post, channel, false)
+		if appErr != nil {
+			return appErr
+		}
+	}
+	return nil
+}
+
+func (a *App) getDisableBotSysadminMessage(user *model.User, userBots model.BotList) string {
+	disableBotsSetting := *a.Config().ServiceSettings.DisableBotsWhenOwnerIsDeactivated
+
+	var printAllBots = true
+	numBotsToPrint := len(userBots)
+
+	if numBotsToPrint > 10 {
+		numBotsToPrint = 10
+		printAllBots = false
+	}
+
+	var message, botList string
+	for _, bot := range userBots[:numBotsToPrint] {
+		botList += fmt.Sprintf("* %v\n", bot.Username)
+	}
+
+	T := utils.GetUserTranslations(user.Locale)
+	infoMessage := T("app.bot.get_disable_bot_sysadmin_message.infoMessage")
```

Comment:
This is probably going to be a total mess to translate. Is there any way we can simplify this? Maybe use some of the [template](https://golang.org/pkg/text/template/) stuff to make it all into one message?

Reply:
I hadn't worked with go templating files, but yes, I see how that would help clean up this code and provide an easier method for creating translations.  I'll look for the docs, but I see there is a `templates/` folder in the server repo for template files.

Reply:
Hey @crspeller, with the current setup, the translation IDs are split into sentences that are constructed based on the logic here. (ie. disableBotsSetting and printAllBots values). The infoMessage translation ID is shared by all messages so it is also abstracted into its own ID. Other thoughts I considered when creating this was keeping the translation ID's simpler instead of creating messages that repeat some of the language.

I looked at the templating options, considering your suggestion to create one message. While it would certainly clean up the code by reducing the Translation to just one ID, that Translation for the message would be rather large and would have to use the templating {{if}}{{else}} logic. Is that preferable?

Reply:
@jfrerich Ya it's better because translators need to see the logic to properly translate.

Reply:
I will simplify the code to call one ID that passes in the necessary parameters and put the logic in the template.  Thanks

Reply:
@crspeller, I just committed a change that pushes all logic inside a single translation message and removes the logic from this function.  The one thing that sticks out as a bit of an eyesore is how much text is inside `"id": "app.bot.get_disable_bot_sysadmin_message"` within `i18n/en.json`.  I don't think it would look as bad if we could format the `json string` into multiple lines but that isn't an option with `json strings`